CMakeLists.txt

```diff
@@ -273,6 +273,8 @@ else()
     target_compile_definitions(kvrocks_objs PUBLIC METADATA_ENCODING_VERSION=0)
 endif()
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror=unused-parameter")
```

Comment:
Setting `CMAKE_CXX_FLAGS` globally is an anti-pattern in CMake.
Maybe the same effect can be achieved with 
```cmake
target_compile_options(kvrocks_objs PUBLIC -Werror=unused-parameter)
```